# Task 1: Permutations and Combinations

By conducting this experiment, we intend to prove whether the person’s ability to identify the preparation method of the tea cups is beyond random chance. If the person can consistently and accurately identify all six milk-first cups, the evidence would support the claim that they possess a special sensory ability. Conversely, if their success rate aligns with what would be expected by random guessing, the claim would lack statistical support.

In summary, this experiment serves as a practical application of hypothesis testing, allowing us to objectively evaluate extraordinary claims using statistical evidence.

*By setting up this experiment, we can:*

- **Quantify Confidence**: Determine the likelihood that the person’s success is due to random chance versus a genuine ability.
+ **Statistical Significance**: Establish whether the results are statistically significant, meaning the observed outcome is unlikely to have occurred under the null hypothesis.
* **Objective Evaluation**: Provide an objective framework to evaluate claims of special abilities, ensuring decisions are based on evidence and probability rather than subjective judgment.

### Permutations
    
A permutation is an ordering of its elements.

The set `{a, b, c}` has the permutations `abc`, `acb`, `bac`, `bca`, `cab`, and `cba`.

### Combinations

A combination is a selection of `k` elements from a set of size `n`.

In combinations, order does not matter and the same element cannot be selected twice.

The combinations of size 2 from the set `{a, b, c}` are `{a, b}`, `{a, c}`, and `{b, c}`.

In [1]:
import math

1 2 3 4 5 6 7 8

Calculate the number of ways to choose **k** elements from a set of **n** elements where the order of the chosen elements does not matter and no element is selected more than once.

In [5]:
# Number of cups of tea in total.
no_cups = 8

# Number of cups of tea with milk first.
no_cups_milk_first = 4

# Number of cups of tea with tea first.
no_cups_tea_first = 4

How many possible combinations are there to shuffle the 8 cups?


In [8]:
# Number of ways of selecting four cups from eight.
ways = math.comb(no_cups, no_cups_milk_first)

# Show.
ways

70

1 2 3 4 5 6 7 8

: _ _ _ _

==> COMBINATION

In [10]:
# Total number of ways of selecting four cups from eight, keeping them in order.
ways_ordered = 8 * 7 * 6 * 5

# Show.
ways_ordered

1680

1680 possible selections

Do you think order is important?

4 8 6 5

8 6 4 5

: _ _ _ _

==> PERMUTATION

In [11]:
# No of ways of shuffling four cups.
no_shuffles = 4 * 3 * 2 * 1

# Show.
no_shuffles

24

24 ways of ordering 4 elements

In [13]:
# No of combinations.
no_combs = ways_ordered // no_shuffles

# Show.
no_combs

70

## Formula

In mathematics, the [`math.comb()`](https://docs.python.org/3/library/math.html#math.comb') function is called the *choose* function.

It is written and calculated as

$\binom{n}{k} = \frac{n!}{k!(n-k)!}$

**Factorial**  

**n!** = the multiplication of all positive integers, say "n", that will be smaller than or equivalent to n is known as the factorial.

**Number of cups of tea in total.**  

**n** = 8

**Number of cups of tea with milk/tea in first.**  

**k** = 4

## [math.factorial](https://docs.python.org/3/library/math.html#math.factorial)

In [20]:
# Number of cups
n = 8

# Number of cups of tea with milk/tea first.
k = 4

# Factorial on number of cups.
fact_n = math.factorial(n)

# Factorial of number of cups of tea with milk/tea first.
fact_k = math.factorial(k)

# Factorial of  number of cups - number of cups of tea with milk/tea first.
diff_n_k =  n-k
fact_diff_n_k = math.factorial(diff_n_k)

# Applying formula: COMBINATION
# No of ways of selecting k objects from n without replacement and without order.
comb = fact_n // (fact_k * fact_diff_n_k)

# Show result.
comb

70

```  
n! = 8 x 7 x 6 x 5 x 4 x 3 x 2 x 1  
k! = 4 x 3 x 2 x 1  
(n - k)! = 4 x 3 x 2 x 1  
n!/(n-k) = (8 x 7 x 6 x 5 x 4 x 3 x 2 x 1) / (4 x 3 x 2 x 1) = 8 x 7 x 6 x 5  
```  

# Task 2: numpy's Normal Distribution

# Task 3: t-Test Calculation

# Task 4: ANOVA